SillyBusinessInc/Moldbreaker
# Feat/#220 add particle systems to enemy

## Purpose of this PR:
This PR adds the walking particles from the player to the Melee Enemy.
It also adds a particle system that plays after the death animation of this Enemy.

The Enemy now also flashes red on hit.

## How to test:
Test this in any scene with the Melee Enemy. And observe it walking, check if there are particle effects.
Try hitting it, check if it flashed red.
upon death check for particle effect to go off after the animation has finished.

https://github.com/user-attachments/assets/2493e697-446f-468d-9b2c-15d4592ab587


### links: 
closes #220
